src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateResults = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    transformState(stateCopy, action);
+
+    stateResults.push({ ...stateCopy });
+  }
+
+  return stateResults;
+}
+
+/**
+ * @param {Object} state
+ * @param {Object} action
+ */
+function transformState(state, action) {
+  switch (action.type) {
+    case 'addProperties':
+      Object.assign(state, action.extraData);
+      break;
```

Comment:
It's a bad practice to mutate parameters. Instead, create a copy and return it from the function.